action/logging/pretty_handler.go

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: MIT
+
+// Package logging for dealing with logging, log settings and log format
+//
+//	Inspiration:
+//		https://github.com/golang/example/tree/master/slog-handler-guide
+//		https://dusted.codes/creating-a-pretty-console-logger-using-gos-slog-package
+package logging
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"runtime"
+	"sync"
+)
+
+var (
+	errInnerHandle          = errors.New("calling inner handler's Handle failed")
+	errInnerHandleUnmarshal = errors.New("unmarshal output on inner handler's Handle failed")
+)
+
+// PrettyHandler is for prettier logs
+type PrettyHandler struct {
+	baseHandler slog.Handler
+	opts        *slog.HandlerOptions
+	mutex       *sync.Mutex
+	buffer      *bytes.Buffer
+	indent      bool
+	output      io.Writer
+}
+
+// NewPrettyHandler returns a PrettyHandler
+func NewPrettyHandler(output io.Writer, opts *slog.HandlerOptions, indent bool) *PrettyHandler {
```

Comment:
I'm not too happy with the option passing here. Having an option struct and two single options makes me wonder why we don't have a HandlerOptions struct here which inherits the HandlerOptions struct from slog. An even better solution imo would be to use the functional options pattern here as it allows setting same defaults without the user needing to know all input options. Also makes sanitization obsolete inside this function (e.g. checking the contents of opts fields against nil.

Reply:
Functional options pattern seems like a good option. Will do that.